neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+        """
+        The Allego Neuronexus reader for the `xdat` file format
+
+        Parameters
+        ----------
+        filename: str | Path, default: ''
+            The filename of the metadata file should end in .xdat.json
+
+        Notes
+        -----
+        * The format involves 3 files:
+            * The *.xdat.json metadata file
+            * The *_data.xdat binary file of all raw data
+            * The *_timestamps.xdat binary file of the timeestamp data
+        From the metadata the other two files are located within the same directory
+        and loaded.
+
+        * The metadata is stored as the metadata attribute for individuals hoping
+        to extract probe information, but neo itself does not load any of the probe
+        information
+
+        Examples
+        --------
+        >>> from neo.rawio import NeuronexusRawIO
+        >>> reader = NeuronexusRawIO(filename='abc.xdat.json')
+        >>> reader.parse_header()
+        >>> raw_chunk = reader.get_analogsignal_chunk(block_index=0
+                seg_index=0,
+                stream_index=0)
+        # this isn't necessary for this reader since data is stored as float uV, but
+        # this is included in case there is a future change to the format
+        >>> float_chunk = reader.rescale_signal_raw_to_float(raw_chunk, stream_index=0)
+
+        """
+
+        BaseRawIO.__init__(self)
+
+        if not Path(filename).is_file():
+            raise FileNotFoundError(f"The metadata file {filename} was not found")
+        if Path(filename).suffix != ".json":
+            raise NeoReadWriteError(
+                f"The json metadata file should be given, filename entered is {Path(filename).stem}"
+            )
+        meta_filename = Path(filename)
+        binary_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_data.xdat")
+
+        if not binary_file.exists() and not binary_file.is_file():
+            raise FileNotFoundError(f"The data.xdat file {binary_file} was not found. Is it in the same directory?")
+        timestamp_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_timestamp.xdat")
+        if not timestamp_file.exists() and not timestamp_file.is_file():
+            raise FileNotFoundError(
+                f"The timestamps.xdat file {timestamp_file} was not found. Is it in the same directory?"
+            )
+
+        self.filename = filename
+        self.binary_file = binary_file
+        self.timestamp_file = timestamp_file
+
+    def _source_name(self):
+        # return the metadata filename only
+        return self.filename
+
+    def _parse_header(self):
+
+        # read metadata
+        self.metadata = self.read_metadata(self.filename)
+
+        # Collect information necessary for memory map
+        self._sampling_frequency = self.metadata["status"]["samp_freq"]
+        self._n_samples, self._n_channels = self.metadata["status"]["shape"]
+        # Stored as a simple float32 binary file
```

Comment:
Dude, where did you get the specification or how did you figured this one out?

Reply:
So you tried the different types until it worked?

Reply:
Yes.... And I looked through comments in some thread where someone thought it was float32. So I iterated a little bit and found things that looked correct. For this to have no gain or offset it basically had to be some type of float.